File: src/problems/Problem85.java
package problems;

public class Problem85 {
	public static void main(String[] args) {
		System.out.println(solution());
	}

	public static int solution() {
		int beste = 2000000;
		int areal = Integer.MAX_VALUE;
		for (int x = 0; x < 2500; ++x) {
			for (int n = 0; n < 2500; ++n) {
				int sum = ((x*(x+1))/2)*(n*(n+1))/2;
				if (Math.abs(sum-2000000) < beste) {
					beste = Math.abs(sum-2000000);
					areal = x*n;
				}
			}
		}
//		System.out.println(areal);
		return areal;
	}
}

File: src/scala/Problem85.scala
package scala

object Problem85 extends App {
  val candidates = for (x <- 1 to 100; y <- 1 to 100) yield (x*y, ((x*(x+1))/2)*(y*(y+1))/2)
  println(candidates.minBy(x => Math.abs(2000000-x._2))._1)
}
